src/aws-cpp-sdk-transfer/source/transfer/TransferManager.cpp

```diff
@@ -1147,6 +1162,51 @@ namespace Aws
                 handle->UpdateStatus(DetermineIfFailedOrCanceled(*handle));
                 TriggerTransferStatusUpdatedCallback(handle);
             }
+
+            if (handle->GetPendingParts().empty() && m_transferConfig.validateChecksums && !handle->GetChecksum().empty()) {
+                uint64_t combinedChecksum = 0;
```

Comment:
i think we want this logic to go [here](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-transfer/source/transfer/TransferManager.cpp#L1238-L1250) when we change the download status to complete, we can also change it to a error if it fails.